common/archiver/archivalMetadata.go

```diff
@@ -73,25 +74,70 @@ const (
 
 // NewArchivalMetadata constructs a new ArchivalMetadata
 func NewArchivalMetadata(
+	dc *dynamicconfig.Collection,
 	historyStatus string,
 	historyReadEnabled bool,
 	visibilityStatus string,
 	visibilityReadEnabled bool,
 	domainDefaults *config.ArchivalDomainDefaults,
 ) ArchivalMetadata {
+	historyConfig := NewArchivalConfig(
+		historyStatus,
+		historyReadEnabled,
+		domainDefaults.History.Status,
+		domainDefaults.History.URI,
+	)
+	if historyConfig.ClusterConfiguredForArchival() {
+		// Only check dynamic config when archival is enabled in static config.
+		// Dynamic archival config is accessed once on cluster startup than never accessed again.
+		// This is done so as to keep archival status and and the initialization of archiver.Provider in sync.
+		// TODO: Once archival pause is implemented archival config can be made truly dynamic.
+		dynamicHistoryStatusString := dc.GetStringProperty(dynamicconfig.HistoryArchivalStatus, historyStatus)()
+		dynamicHistoryStatus, err := getClusterArchivalStatus(dynamicHistoryStatusString)
+		if err != nil {
+			panic(err)
```

Comment:
I am a little scared about panic based on a dynamic config value. It seems fine to panic based on static config because there is nothing you can do in that case. But here if someone changes dynamic config to invalid value then restarts the cluster all the nodes will start panic on startup and we will have an outage. Can we instead just default to disabled if we cannot read archival config. 

If someone meant to enable but actually entered "emabled" (or some other mispelling) its reasonable that archival will stay disabled.

Reply:
Sorry I should have thought about that earlier.